agent/src/main/java/com/cmclinnovations/agent/service/GetService.java

```diff
@@ -235,14 +237,49 @@ public ResponseEntity<StandardApiResponse<?>> getConceptMetadata(String conceptC
     // Note that all concept metadata will never be stored in Ontop and will require
     // the special property paths
     Queue<SparqlBinding> results = this.kgService.query(query, SparqlEndpointType.BLAZEGRAPH);
+    List<Map<String, Object>> resultItems;
     if (results.isEmpty()) {
       LOGGER.info(
           "Request has been completed successfully with no results!");
+      resultItems = new ArrayList<>();
     } else {
       LOGGER.info(SUCCESSFUL_REQUEST_MSG);
+      SparqlBinding rootClassBinding = results.stream()
+          .filter(
+              binding -> binding.getFieldValue("type").equals(conceptClass) && binding.getFieldValue("parent") != null)
+          .findFirst()
+          .orElse(null);
+      if (rootClassBinding != null) {
+        Set<String> rootParentClasses = Arrays.stream(rootClassBinding.getFieldValue("parent").split("\\s"))
+            .collect(Collectors.toSet());
+        rootParentClasses.add(conceptClass);
+        resultItems = results.stream()
+            .map(binding -> {
+              Map<String, Object> bindMappings = binding.get();
+              SparqlResponseField parentField = (SparqlResponseField) bindMappings.get("parent");
+              String parentValue = parentField.value();
+              if (parentValue != null) {
+                String[] currentParentClasses = parentValue.split("\\s+");
+                List<SparqlResponseField> remainingClasses = Arrays.stream(currentParentClasses)
+                    .filter(s -> !rootParentClasses.contains(s))
+                    .map(
+                        s -> new SparqlResponseField(parentField.type(), s, parentField.dataType(), parentField.lang()))
+                    .toList();
+                if (remainingClasses.isEmpty()) {
+                  bindMappings.remove("parent");
+                } else {
+                  // Typically, there are only one parent class
+                  bindMappings.put("parent", remainingClasses.get(0));
```

Comment:
Probably worth us having a discussion about this in case we can remove this assumption/requirement.

Reply:
Misunderstanding from me